tests/nightly/test_large_array.py

```diff
@@ -279,6 +272,13 @@ def test_diag():
     assert_almost_equal(r.asnumpy(), np.diag(a_np, k=k))
 
 
+def test_softmax():
+    input_data = mx.nd.ones((SMALL_Y, LARGE_X))
+    true_output = np.full((SMALL_Y, LARGE_X), (1 / SMALL_Y))
+    output = nd.softmax(input_data, axis=0)
```

Comment:
when you test this way, the softmax is operating only on one dimension which does not exceed 2^32, right?

Reply:
yes

Reply:
But shouldn't we test softmax over a vector larger than 2^32?

Reply:
The test fails

Reply:
fine. let's leave that as TODO in the next phase to support large int in one dimension.